tidy3d/components/device/simulation.py

```diff
@@ -881,20 +1014,21 @@ def _get_structure_source_plot_params(
         if alpha is not None:
             plot_params = plot_params.copy(update={"alpha": alpha})
 
-        if isinstance(source, UniformHeatSource):
-            rate = source.rate
-            delta_rate = rate - source_min
-            delta_rate_max = source_max - source_min + 1e-5
-            rate_fraction = delta_rate / delta_rate_max
-            cmap = cm.get_cmap(HEAT_SOURCE_CMAP)
-            rgba = cmap(rate_fraction)
-            plot_params = plot_params.copy(update={"edgecolor": rgba})
+        if isinstance(source, HeatSource):
+            rate = self._safe_float_conversion(source.rate)
+            if rate is not None:
```

Comment:
what if rate is None? should we warn? or is it ok?

Reply:
I think it's OK. When it's None, it just doesn't display the colormap for sources